tls/tests/test_local.py

```diff
@@ -0,0 +1,171 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import pytest
+
+from datadog_checks.base import ConfigurationError
+from datadog_checks.tls import TLSCheck
+
+
+def test_no_cert(instance_local_no_cert):
+    c = TLSCheck('tls', {}, [instance_local_no_cert])
+
+    with pytest.raises(ConfigurationError):
+        c.check(None)
+
+
+def test_no_server_hostname(instance_local_no_server_hostname):
+    c = TLSCheck('tls', {}, [instance_local_no_server_hostname])
+
+    with pytest.raises(ConfigurationError):
+        c.check(None)
+
+
+def test_not_found(aggregator, instance_local_not_found):
+    c = TLSCheck('tls', {}, [instance_local_not_found])
+    c.check(None)
+
+    aggregator.assert_service_check(c.SERVICE_CHECK_CAN_CONNECT, count=0)
+    aggregator.assert_service_check(c.SERVICE_CHECK_VERSION, count=0)
+    aggregator.assert_service_check(c.SERVICE_CHECK_VALIDATION, status=c.CRITICAL, tags=c._tags, count=1)
```

Comment:
Note that by using c._tags you don't really tests anything super useful. You could remove all tags from the check and the test would still pass.

Reply:
It's testing tags are part of the payload. The testing of what the tags should be happens in test_config.py